### DISCUSSION TITLE
Calendar and CalendarItem DisplayDate

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#8197] [General]
DisplayDate is defined as not nullable

https://github.com/AvaloniaUI/Avalonia/blob/194ddc5525b0d42882aa029f705481d264208dfc/src/Avalonia.Controls/Calendar/Calendar.cs#L732-L737

But in some part used as nullable

https://github.com/AvaloniaUI/Avalonia/blob/194ddc5525b0d42882aa029f705481d264208dfc/src/Avalonia.Controls/Calendar/Calendar.cs#L341

https://github.com/AvaloniaUI/Avalonia/blob/194ddc5525b0d42882aa029f705481d264208dfc/src/Avalonia.Controls/Calendar/Calendar.cs#L1281

https://github.com/AvaloniaUI/Avalonia/blob/194ddc5525b0d42882aa029f705481d264208dfc/src/Avalonia.Controls/Calendar/Calendar.cs#L1329

https://github.com/AvaloniaUI/Avalonia/blob/194ddc5525b0d42882aa029f705481d264208dfc/src/Avalonia.Controls/Calendar/Calendar.cs#L1900

What is the correct behavior?

**Comment by maxkatz6** [comment]
By current API surface it's not nullable. So these checks can be removed.

**Reply by robloo** [reply]
First, this was carryover from very old Microsoft code base. So the property definition is not matchung Microsoft design intent which is usually heavily reviewed. 

However, in this case original API is must be correct I think. The DateTime struct has always been a value type so nullability is explicit. Moreover, of course you can have no selected date in a calendar. This is internally represented by null.

**Reply by maxkatz6** [reply]
Yeah. But I assume context of the question was how to solve the warnings. Without changing any behavior.

**Reply by robloo** [reply]
Yea, I guess I would just make the opposite assumption. The Avalonia property definition is wrong and was incorrectly ported:

https://github.com/AvaloniaUI/Avalonia/blob/194ddc5525b0d42882aa029f705481d264208dfc/src/Avalonia.Controls/Calendar/Calendar.cs#L751-L756

**Reply by workgroupengineering** [reply]
> Yeah. But I assume context of the question was how to solve the warnings. Without changing any behavior.

Partly.

Since we are in a period of breaking changes.
I wanted to understand what the core team's vision is before proposing a PR given discussion #8113.
I personally have the same opinion as @robloo.
